### ISSUE TITLE
No table names shown in Dataset creation window / SQL Lab for Apache Kyuubi

### ISSUE THREAD

**Issue by Miosp** [apache/superset#28743]
### Bug description

I'm trying to connect Apache Kyuubi to superset. I settled on hive connector through pyhive (the issues with mySQL and trino connectors are even worse). However I'm facing an issue where my tables are not shown in SQL lab. I can still construct a SQL query and get the data manually, but for the sake of correctness the feature should work. 

### How to reproduce the bug

1. Install Superset via helm with minimal configuration (just set the SUPERSET_SECRET_KEY, which is required for the chart)
2. Connect kyuubi with uri: hive://<kyuubi_ip>:<kyuubi_spark_thrift_port>/default
3. In the Data Creation window / SQL Lab choose the name of the connection and schema default
4. See that instead of your tables you see view 'default'

### Screenshots/recordings

![Zrzut ekranu 2024-05-28 225447](https://github.com/apache/superset/assets/30725330/d82671f3-e433-4346-b95e-aa7882614266)

### Superset version

4.0.1

### Python version

3.9

### Node version

I don't know

### Browser

Chrome

### Additional context

When i click the "refresh" button next to table list in superset, in kyuubi's spark driver log i can see following SQL queries executed:
USE `default`
SHOW VIEWS IN `default`
USE `default`
USE `default`
SHOW VIEWS IN `default`

There are no errors in superset's logs.

### Checklist

- [X] I have searched Superset docs and Slack and didn't find a solution to my problem.
- [X] I have searched the GitHub issue tracker and didn't find a similar bug report.
- [X] I have checked Superset's logs for errors and if I found a relevant Python stacktrace, I included it here as text in the "additional context" section.

**Comment by tyagiaksh** [comment]
@dosu still not able to resolve above issue

**Comment by tyagiaksh** [comment]
@Miosp have you resolved this issue?

**Comment by Miosp** [comment]
No, in order to overcome this i had to create a source from an SQL query and use that. SQL queries execute just fine.

**Comment by tyagiaksh** [comment]
yes I have also solved using that

**Comment by rusackas** [comment]
Hey folks. I'll close this since it has gone inactive for 250+ days. If there are docs that should be added, or improvements that can be made here, we're VERY open to PRs, but otherwise, I guess we'll call this case closed. Happy to revisit if needed.
